### PR TITLE
Feat: Check def binding to a component

### PR DESCRIPTION
**Pull request by wangyikewxgm**
1. addon check definition whether binding a  component, if this component not exist don't apply it.
2. apiserver show readme.md
3. apiserver show cue definition

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/addon/utils.go [hunk 1]
```diff
@@ -52,12 +52,17 @@ const (
 	traitMapKey               = "trait"
 	wfStepMapKey              = "wfStep"
 	policyMapKey              = "policy"
+	annotationBindCompKey     = "addon.oam.dev/ignore-without-component"
```
Comment (wonderflow) [inline pkg/addon/utils.go:55]:
```suggestion        
	annotationBindCompKey     = "addon.oam.dev/bind-component"
```

Move it along with other annotations

File: pkg/addon/addon.go [hunk 2]
```diff
@@ -1100,9 +1108,12 @@ func (h *Installer) dispatchAddonResource(addon *InstallPackage) error {
 
 	for _, def := range defs {
 		addOwner(def, app)
-		err = h.apply.Apply(h.ctx, def, apply.DisableUpdateAnnotation())
-		if err != nil {
-			return err
+		if checkObjectBindingComponent(*def, *app) {
```
Comment (wonderflow) [inline pkg/addon/addon.go:1111]:
```suggestion
		if !checkObjectBindingComponent(*def, *app) {
		 continue
		}
```

File: pkg/addon/addon.go [hunk 3]
```diff
@@ -1123,10 +1134,12 @@ func (h *Installer) dispatchAddonResource(addon *InstallPackage) error {
 	}
 
 	for _, o := range auxiliaryOutputs {
-		addOwner(o, app)
-		err = h.apply.Apply(h.ctx, o, apply.DisableUpdateAnnotation())
-		if err != nil {
-			return err
+		if checkObjectBindingComponent(*o, *app) {
```
Comment (wonderflow) [inline pkg/addon/addon.go:1137]: use `continue` like above

File: pkg/addon/utils.go [hunk 4]
```diff
@@ -474,3 +479,20 @@ func produceDefConflictError(conflictDefs map[string]string) error {
 	errorInfo += "if you want override them, please use argument '--override-definitions' to enable \n"
 	return errors.New(errorInfo)
 }
+
+// checkObjectRelatedComponent will check the ready-to-apply object(def or auxiliary outputs) whether bind to a component
+// if the target component not exist, return false.
+func checkObjectBindingComponent(u unstructured.Unstructured, app v1beta1.Application) bool {
```
Comment (wonderflow) [inline pkg/addon/utils.go:485]:
```suggestion
func checkBondComponentExist(u unstructured.Unstructured, app v1beta1.Application) bool {
```

File: pkg/addon/utils.go [hunk 5]
```diff
@@ -474,3 +479,20 @@ func produceDefConflictError(conflictDefs map[string]string) error {
...
+	comp, existKey := u.GetAnnotations()[annotationBindCompKey]
+	if !existKey {
+		// if an object(def or auxiliary outputs ) binding no components return true
+		return true
+	}
+	for _, component := range app.Spec.Components {
+		if component.Name == comp {
+			// find the match component, return ture
```
Comment (wonderflow) [inline pkg/addon/utils.go:493]:
```suggestion
			// the bond component exists, return ture
```

**Review by wonderflow — APPROVED** [review]

**Review by FogDong — APPROVED** [review]

**Review by barnettZQG — APPROVED** [review]
LGTM

**Pull request merged**